File: go/conditional_statements.go
package main

import "fmt"

func main() {
	// If-else statement
	number := 7

	if number < 5 {
		fmt.Println("Number is less than 5")
	} else if number > 5 {
		fmt.Println("Number is greater than 5")
	} else {
		fmt.Println("Number is equal to 5")
	}

	// Switch statement
	day := 3
	switch day {
	case 1:
		fmt.Println("Monday")
	case 2:
		fmt.Println("Tuesday")
	case 3:
		fmt.Println("Wednesday")
	case 4:
		fmt.Println("Thursday")
	case 5:
		fmt.Println("Friday")
	case 6, 7:
		fmt.Println("Weekend")
	default:
		fmt.Println("Invalid day")
	}

	// Switch without expression (like if-else)
	age := 25
	switch {
	case age < 18:
		fmt.Println("Minor")
	case age >= 18 && age < 65:
		fmt.Println("Adult")
	default:
		fmt.Println("Senior")
	}

	// If with short statement
	if score := 85; score >= 70 {
		fmt.Println("Passing grade")
	} else {
		fmt.Println("Failing grade")
	}

	// Error handling with if
	result, err := divide(10, 2)
	if err != nil {
		fmt.Println("Error:", err)
	} else {
		fmt.Printf("Result: %f\n", result)
	}
}

// Helper function for error handling example
func divide(a, b float64) (float64, error) {
	if b == 0 {
		return 0, fmt.Errorf("division by zero")
	}
	return a / b, nil
}

File: typescript/conditional_statements.ts
// Conditional statements in TypeScript

// Basic if-else
const number: number = 7;

if (number < 5) {
    console.log("Number is less than 5");
} else if (number > 5) {
    console.log("Number is greater than 5");
} else {
    console.log("Number is equal to 5");
}

// Switch statement
const day: number = 3;
switch (day) {
    case 1:
        console.log("Monday");
        break;
    case 2:
        console.log("Tuesday");
        break;
    case 3:
        console.log("Wednesday");
        break;
    case 4:
        console.log("Thursday");
        break;
    case 5:
        console.log("Friday");
        break;
    case 6:
    case 7:
        console.log("Weekend");
        break;
    default:
        console.log("Invalid day");
}

// Ternary operator
const age: number = 20;
const canVote: string = age >= 18 ? "Yes" : "No";
console.log(`Can vote? ${canVote}`);

// Nullish coalescing
const userName: string | null = null;
const displayName: string = userName ?? "Anonymous";
console.log(`User: ${displayName}`);

// Optional chaining with conditional
type User = {
    name: string;
    address?: {
        street?: string;
        city?: string;
    };
};

const user: User = {
    name: "John",
    address: {
        city: "New York"
    }
};

if (user.address?.street) {
    console.log("Street address available");
} else {
    console.log("No street address");
}
